src/trace/context/extractors/sqs.ts

```diff
@@ -3,18 +3,29 @@ import { EventTraceExtractor } from "../extractor";
 import { TracerWrapper } from "../../tracer-wrapper";
 import { logDebug } from "../../../utils";
 import { SpanContextWrapper } from "../../span-context-wrapper";
+import { XrayService } from "../../xray-service";
 
 export class SQSEventTraceExtractor implements EventTraceExtractor {
   constructor(private tracerWrapper: TracerWrapper) {}
 
   extract(event: SQSEvent): SpanContextWrapper | null {
-    const headers = event?.Records?.[0]?.messageAttributes?._datadog?.stringValue;
-    if (headers === undefined) return null;
-
     try {
-      const traceContext = this.tracerWrapper.extract(JSON.parse(headers));
-      if (traceContext === null) return null;
+      var prepared_headers;
+      const headers = event?.Records?.[0]?.messageAttributes?._datadog?.stringValue;
+      if (headers !== undefined) {
+        prepared_headers = JSON.parse(headers);
+      } else {
+        if (event?.Records?.[0]?.attributes?.AWSTraceHeader) {
+          prepared_headers = XrayService.extraceDDContextFromAWSTraceHeader(event.Records[0].attributes.AWSTraceHeader);
+        }
+      }
 
+      if (!prepared_headers) return null;
+      const traceContext = this.tracerWrapper.extract(prepared_headers);
+      if (traceContext === null) {
+        logDebug("Unable to extract the injected trace context from event");
```

Comment:
Let's be more specific with this message

Reply:
Not sure I understand it correctly. Do you mean something like 
`logDebug("Unable to extract the injected trace context from event ${preparedHeaders}");`

Reply:
So, any engineer / users might see the message and not understand where this message comes from since it's pretty generic, maybe we can add something around the lines `Unable to extract the injected trace context from SQS event`?

Reply:
Got it. updated it to ` logDebug("Failed to extract trace context from prepared headers: ${preparedHeaders}");`
`